pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/SumAggregationFunction.java

```diff
@@ -28,13 +30,23 @@
 import org.apache.pinot.core.query.aggregation.groupby.DoubleGroupByResultHolder;
 import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
 import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.roaringbitmap.RoaringBitmap;
 
 
 public class SumAggregationFunction extends BaseSingleInputAggregationFunction<Double, Double> {
   private static final double DEFAULT_VALUE = 0.0;
+  private final boolean _nullHandlingEnabled;
+
+  private final Set<Integer> _groupKeysWithNonNullValue;
```

Comment:
Let's fix this function similarly to the `MIN` and `MAX`: use object-based result holder, and fix `extractAggregationResult()`

Reply:
I just double checked Presto behaviour and Sum returns null not zero:

```
SELECT sum(id) as sum, key FROM (VALUES (null, 1), (null, 1), (null, 2), (1, 3), (null, 3)) AS t(id, key)  GROUP BY key ORDER BY key DESC;
 sum  | key
------+-----
    1 |   3
 NULL |   2
 NULL |   1
```

Updated to be like `MIN` and `MAX`.

Reply:
This one is not fixed. Is there a commit not pushed?

Reply:
yes, I mark resolved as soon as I change it in my local git repo and send the new iteration later.